Python/pylifecycle.c

```diff
@@ -2719,6 +2729,70 @@ init_sys_streams(PyThreadState *tstate)
 }
 
 
+#ifdef __ANDROID__
+#include <android/log.h>
+
+static PyObject *
+android_log_write_impl(PyObject *self, PyObject *args)
+{
+    int prio = 0;
+    char *tag = NULL;
+    char *text = NULL;
+    Py_ssize_t *text_len = 0;
+    if (!PyArg_ParseTuple(args, "iss#", &prio, &tag, &text, &text_len))
```

Comment:
Do we need to allow calling this function with `str` or only with `bytes`? Note that it will fail if `str` is not UTF-8 encodable. If it is not needed, it is preferable to use `y*` instead of `s#`.

Reply:
This function is only called with `bytes`, and because of the previous change, there will be no null bytes. So I've used the format `y`.